skglm/tests/test_group.py

```diff
@@ -123,5 +122,32 @@ def test_vs_celer_grouplasso(n_groups, n_features, shuffle):
     np.testing.assert_allclose(model.coef_, w, atol=1e-5)
 
 
+def test_anderson_acceleration():
+    # VAR: w = rho * w + 1 with |rho| < 1
+    # converges to w_star = 1 / (1 - rho)
+    max_iter, tol = 1000, 1e-9
+    n_features = 2
+    rho = np.array([0.5, 0.8])
+    w_star = 1 / (1 - rho)
+    X = np.diag([2, 5])
+
+    acc = AndersonAcceleration(K=5)
+    n_iter_acc = 0
+    w = np.ones(n_features)
+    Xw = X @ w
+    for i in range(max_iter):
+        w, Xw = acc.extrapolate(w, Xw)
+        w = rho * w + 1
+        Xw = X @ w
+
+        if norm(w - w_star, ord=np.inf) < tol:
+            n_iter_acc = i
+            break
+
+    np.testing.assert_allclose(w, w_star)
+    np.testing.assert_allclose(Xw, X @ w_star)
+    np.testing.assert_array_equal(n_iter_acc, 13)
```

Comment:
I'm sorry, maybe I was not clear here: I agree with you that we should test that AA performs way better than regular iteration. I prefer testing equality because it checks that there is a stringent difference (13 vs 99), that is also made clear to the reader (while 98 vs 99 would still pass the n_iter < n_iter_acc test).

So we also need to test that n_iter (regular) is equal to 99. Otherwise, checking convergence of AA in 13 iterations here does not check that it's faster than regular iterations.